aten/src/ATen/native/ScatterGatherChecks.h

```diff
@@ -38,6 +38,9 @@ static void scatter_gather_dtype_check(
 static void gather_shape_check(const Tensor& self, int64_t dim,
   const Tensor& index, const Tensor& src
 ) {
+  if (index.numel() == 0) {
+      return;
+  }
```

Comment:
Not sure if this is the right place to move this code to. This `if` was a short-circuit in `scatter` and `gather` kernels, right? If so, maybe it makes more sense move this `if` to their IMPL functions.

Reply:
You are right, this should be fixed now.